### ISSUE TITLE
Ensure Wazuh_testing tools and FIM integration tests works as intended in Windows environments

### ISSUE THREAD

**Issue by CarlosRS9** [wazuh/wazuh-qa#867]
# Description

Core team reported us that some FIM integrity tests are failing in Windows enviroments with apparently some sort of random errors.

We need to run FIM integrity tests in a Windows environment and check if there is something wrong in the tests or in the `Wazuh_testing` tools used by those tests.

This issue is related to #858.

**Comment by CarlosRS9** [comment]
# Status Update

We have already run `tier 0` FIM integration tests in a Windows 7 machine both with a Wazuh 3.13 and a 4.0 agent. The results seems to be variable: the number of failed tests ranges from 17 to 21 in every scenario. The results do not seem consistent as some test may or may not fail.

We are currently checking the output of those tests to determine if the problem is with the `wazuh_test`, the tests themselves or the Wazuh Core part.

**Comment by CarlosRS9** [comment]
# Status Update

After checking some of the tests that fails sometimes we conclude for that particular case that were due to a creation event that was not raised or at least was not present in the ossec.log. Additionally, the following message appeared several times in the ossec.log, flooding it:

```
DEBUG: (6319): No configuration found for (file):'c:\program files (x86)\ossec-agent\ossec.log'
```
This may have impact over the tests and should not appear. We need to ensure that the entire `C:/` directory is not being monitored when Syscheck is set to monitor a directory that does not exists.

**Comment by CarlosRS9** [comment]
# Status Update

### No configuration found for (file):'c:\program files (x86)\ossec-agent\ossec.log'

It seems all the error detected yesterday only happened with Whodata. Realtime and scheduled were working perfectly. Those errors appeared because Windows' audit was monitoring the directory in which the ossec.log is located. This is because of an wrong configuration in our testing environments and not related to the tests themselves. Once this was fixed the tests passed without any problems.

### Random errors
We noticed that some "random" errors may ocurr when running the tests repeatedly as Core team reported. In these particular cases we have been able to observe anomalies in the timestamps of the events in the `ossec.log`. As an example, the test modified the system clock from `2020-09-20 02:19` to `2020-09-20 15:19`. However, the events in the ossec.log have timestamps for `00:19`, `13:19` and `15:19`. We need to investigate further why these timestamp changes occur and whether this may be the reason for the test's random failures.

### Syscheck scan

Finally, we have detected an unwanted behaviour related to our system's clock modification. Some tests modifies the system clock to trigger a syscheck scan. At the end of those tests the system clock is restored, going backward the same number of hours the clock had previously advanced. For example, if the actual system time is `2020-09-20 10:00` and we and we change it to `2020-09-20 23:00` a syscheck scan will be trigger because the elapsed time is greater than the `frequency` value. If we reset the system clock again to `2020-09-20 10:00` and then set it again to `2020-09-20 23:00` a new scan will NOT be triggered because syscheck will check if a new scan must be triggered comparing the actual timestamp with the one of the previous scan
and in this example both will be `2020-09-20 23:00`. 

To sum up: altering the system clock too much and going backwards *may* cause some test to fail because the syscheck scan won't be triggered.

**Comment by CarlosRS9** [comment]
# Status Update

We have not been able to replicate the error reported by core team yet. The tests are failing randomly because of the internal clock modifications.

The problem related to syscheck scan and time rollback reported in the previous status update are fixed when the agent is restarted. Since the tests restart the agent after restoring the system internal clock this does not seem to be the cause of the our random test failures.

We are still investigating the causes of the strange timestamps presents in some failed tests.
